UndertaleModLib/Decompiler/Decompiler.cs

```diff
@@ -3695,14 +3755,14 @@ private static BlockHLStatement HLDecompileBlocks(DecompileContext context, ref
                         var x = caseEntries.ElementAt(i);
                         Block temp = x.Key;
 
-                        Block switchEnd = DetermineSwitchEnd(temp, caseEntries.Count > (i + 1) ? caseEntries.ElementAt(i + 1).Key : null, meetPoint);
+                        Block switchEnd = context.GlobalContext.Data.GeneralInfo.BytecodeVersion == 17 ? meetPoint : DetermineSwitchEnd(temp, caseEntries.Count > (i + 1) ? caseEntries.ElementAt(i + 1).Key : null, meetPoint);
```

Comment:
This change most likely will cause degradation on older GMS versions, see [this discussion](https://github.com/krzys-h/UndertaleModTool/pull/1203)

Reply:
and this is the large switch statements fix

Reply:
Yes, I know, but I mean that this fixes them on newer GMS versions, but it should be also tested on _many_ older games (not only Undertale and Deltarune).
Please read the discussion again.

Reply:
yea i did, looks like for the older games then the problem doesnt even exist